suite-common/wallet-core/src/send/sendFormThunks.ts

```diff
@@ -538,14 +539,17 @@ export const enhancePrecomposedTransactionThunk = createThunk<
 
         if (!isCardanoTx(selectedAccount, enhancedPrecomposedTransaction)) {
             if (formValues.rbfParams) {
-                enhancedPrecomposedTransaction.prevTxid = formValues.rbfParams.txid;
-                enhancedPrecomposedTransaction.feeDifference = new BigNumber(
-                    precomposedTransaction.fee,
-                )
-                    .minus(formValues.rbfParams.baseFee)
-                    .toFixed();
-                enhancedPrecomposedTransaction.useNativeRbf = useNativeRbf;
-                enhancedPrecomposedTransaction.useDecreaseOutput = hasDecreasedOutput;
+                (enhancedPrecomposedTransaction as PrecomposedTransactionFinalRbf).prevTxid =
+                    formValues.rbfParams.txid;
+                (enhancedPrecomposedTransaction as PrecomposedTransactionFinalRbf).feeDifference =
+                    new BigNumber(precomposedTransaction.fee)
+                        .minus(formValues.rbfParams.baseFee)
+                        .toFixed();
+                (enhancedPrecomposedTransaction as PrecomposedTransactionFinalRbf).useNativeRbf =
+                    !!useNativeRbf;
+                (
+                    enhancedPrecomposedTransaction as PrecomposedTransactionFinalRbf
+                ).useDecreaseOutput = !!hasDecreasedOutput;
```

Comment:
Not exactly beautiful, but I haven't come with anything better, so consider this a pointless rant.

Reply:
I felt the same.